### PR TITLE
Rename "auto" to "javaagent" in modules

### PR DESCRIPTION
**Pull request by trask**
Another part of #1186

This includes https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1186#issuecomment-706770335:

> Trying out shortening the name `opentelemetry-javaagent-instrumentation-api` --> `opentelemetry-javaagent-api`

But not the shortened package rename yet, because still thinking about https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1186#issuecomment-706824508.

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]
SGTM

**Review by iNikem — APPROVED** [review]
LGTM but one question: why we have to repeat `JAVAagent`? Maybe just `agent`?

**Comment by trask** [comment]
> LGTM but one question: why we have to repeat `JAVAagent`? Maybe just `agent`?

I like changing it here:

* armeria-1.0
  * javaagent -> agent
  * library
  * testing

since `library` in this context doesn't really correspond to package name / artifact name either.

For other places, I'm torn:

#### io.opentelemetry.agent

  * `opentelemetry-agent-cats`
  * `opentelemetry-agent-api`
  * `opentelemetry-agent`
  * `opentelemetry-agent-spi`
  * `opentelemetry-agent-exporters-otlp`
  * `opentelemetry-agent-exporters-jaeger`
  * `opentelemetry-agent-exporters-zipkin`
  * `opentelemetry-agent-exporters-logging`
* Internal components of the agent (maybe we don't need to publish?)
  * `opentelemetry-agent-internal-bootstrap`
  * `opentelemetry-agent-internal-tooling`

I like the shorter `agent`, but also we were trying to avoid confusion with external agents, e.g. OpenTelemetry collector agent.

**Comment by iNikem** [comment]
Does anybody call Otel Collector an "agent"?

**Comment by trask** [comment]
> Does anybody call Otel Collector an "agent"?

Yeah, https://opentelemetry.io/docs/collector/about/

Also, when most ops people hear "agent", they think of out-of-process agent (like the collector agent), as opposed to in-process agent (like a javaagent). Some discussion about this occurred here: https://github.com/open-telemetry/oteps/blob/master/text/0001-telemetry-without-manual-instrumentation.md

**Comment by iNikem** [comment]
Let it be "javaagent" then

File: docs/contributing/javaagent-jar-components.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ module
 Available in the bootstrap class loader:
 
 - `io/opentelemetry/javaagent/bootstrap/` - contains the `javaagent-bootstrap` module
-- `io/opentelemetry/instrumentation/auto/api/` - contains the `auto-api` module
+- `io/opentelemetry/javaagent/api/` - contains the `javaagent-api` module
```
Comment (mateuszrzeszutek) [inline docs/contributing/javaagent-jar-components.md:85]: Currently it's `io/opentelemetry/javaagent/instrumentation/api/`

Comment (trask) [reply docs/contributing/javaagent-jar-components.md]: thanks! fixed

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**